frontend/src/modules/SimulationTimeSeriesSensitivity/simulationTimeSeriesChart/chart.tsx

```diff
@@ -1,89 +1,114 @@
 
 import { TimeSeriesPlotlyTrace } from "./traces";
 
-export type timeSeriesChartProps = {
-    traceDataArr: TimeSeriesPlotlyTrace[];
-    onHover?: (date: string) => void;
+export type HoverInfo = {
+    timestampUtcMs: number;
+    realization?: number;
+    shiftKeyIsDown?: boolean;
+};
 
+export type TimeSeriesChartProps = {
+    traceDataArr: TimeSeriesPlotlyTrace[];
+    activeTimestampUtcMs?: number;
+    hoveredTimestampUtcMs?: number;
+    onHover?: (hoverData?: HoverInfo) => void;
+    onClick?: (timestampUtcMs: number) => void;
     height?: number | 100;
     width?: number | 100;
 };
 
-export const TimeSeriesChart: React.FC<timeSeriesChartProps> = (props) => {
-    const { height, width, traceDataArr } = props;
-    const [activeTimestamp, setActiveTimestamp] = useState<string | null>(null);
-    const [hoverActive, setHoverActive] = useState<boolean>(true);
-    const handleClick = () => {
-        setHoverActive(!hoverActive);
-    };
-    const handleHover = (e: PlotHoverEvent) => {
-        if (hoverActive && e.xvals.length > 0 && typeof e.xvals[0]) {
-            // workbenchServices.publishGlobalData("global.hoverTimestamp", { timestamp: e.xvals[0] as number });
-            setActiveTimestamp(e.xvals[0] as string);
-            if (props.onHover) {
-                props.onHover(e.points[0].x as string);
+export const TimeSeriesChart: React.FC<TimeSeriesChartProps> = (props) => {
+    function handleClick(e: PlotMouseEvent) {
+        const clickedPoint: PlotDatum = e.points[0];
+        if (!clickedPoint || !props.onClick) {
+            return;
+        }
+
+        if (clickedPoint.pointIndex >= 0 && clickedPoint.pointIndex < clickedPoint.data.x.length) {
+            const timestampUtcMs = clickedPoint.data.x[clickedPoint.pointIndex];
+            if (typeof timestampUtcMs === "number") {
+                props.onClick(timestampUtcMs);
             }
         }
-        const curveData = e.points[0].data as TimeSeriesPlotlyTrace;
-        if (typeof curveData.realizationNumber === "number") {
-            // setHighlightRealization(curveData.realizationNumber);
-            // workbenchServices.publishGlobalData("global.hoverRealization", {
-            //     realization: curveData.realizationNumber,
-            // });
+    }
+
+    function handleHover(e: PlotHoverEvent) {
+        const hoveredPoint: PlotDatum = e.points[0];
+        if (!hoveredPoint || !props.onHover) {
+            return;
         }
-    };
-    const layout: Partial<Layout> = {
-        width: width,
-        height: height,
 
-        xaxis: {
-            type: "category",
-        },
+        if (hoveredPoint.pointIndex >= 0 && hoveredPoint.pointIndex < hoveredPoint.data.x.length) {
+            const timestampUtcMs = hoveredPoint.data.x[hoveredPoint.pointIndex];
+            if (typeof timestampUtcMs === "number") {
+                let maybeRealizationNumber: number | undefined;
```

Comment:
Detail, but maybe rather use an adjective like `contingent`, `potential` or `possible` instead of an adverb.

Reply:
A detail indeed, I think the term "maybe" snuck into me from working with NodeJS' native api and I kind of like it. Not quite the same but I feel it communicates well.

Reply:
In combination with a function, it makes grammatically sense to use an adverb (e.g. `maybeLoad...`) since it's in combination with a verb, here it is in combination with a noun and ought therefore be an adjective. However, not worth a big discussion. I do only think if this concept of naming possibly undefined variables might spread, we might as well use correct grammar 😛